### PR TITLE
README: Provide one.

### PR DESCRIPTION
**Pull request by cpu**
This adds what I hope is a pretty good README.

I also deleted the `Makefile` and the git hook business. Nobody is using that anymore and its pretty jank.

Resolves https://github.com/letsencrypt/ct-woodpecker/issues/58

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,11 +10,368 @@
...
+  * `"failed-to-verify-proof"` for when a returned STH consistency proof can't
+  be validated.
+
+* Possible `cert_submit_results` `status` values are:
+  * `"fail"` for failed submissions.
+  * `"ok"` for successful submissions.
+
+* `cert_submit_results` will have a true `precert` label when the submission was a precert.
+
+* `cert_submit_results` will have a true `duplicate` label when the submission was a resubmission of a previously submitted cert/precert.
```
Comment (jsha) [inline README.md]:
```suggestion
* `cert_submit_results` will have a `duplicate="true"` label when the submission was a resubmission of a previously submitted cert/precert.
```

File: README.md [hunk 2]
```diff
@@ -10,11 +10,368 @@
...
+  * `"equal-treesize-inequal-hash"` for when two STH's have the same treesize and different hashes.
+  * `"failed-to-get-proof"` for when an error occurs fetching the consistency proof.
+  * `"failed-to-verify-proof"` for when a returned STH consistency proof can't
+  be validated.
+
+* Possible `cert_submit_results` `status` values are:
+  * `"fail"` for failed submissions.
+  * `"ok"` for successful submissions.
+
+* `cert_submit_results` will have a true `precert` label when the submission was a precert.
```
Comment (jsha) [inline README.md]:
```suggestion
* `cert_submit_results` will have a `precert="true"` label when the submission was a precert.
```

File: README.md [hunk 3]
```diff
@@ -10,11 +10,368 @@
...
+* Possible `cert_storage_failures` `type` values are:
+  * `"marshalling"` for failures to marshal a returned SCT for storage.
+  * `"storing"` for failures to insert the cert/SCT into the DB.
+
+* Possible `inclusion_checker_errors` `type` values are:
+  * `"getIndex"` for failures to get the current stored tree index from the DB.
+  * `"getUnseen"` for failures to find unseen certs/SCTs in the DB.
+  * `"getSTH"` for failures to fetch an STH to determine entries needing to be
+  fetched.
+  * `"getEntires"` for failures to get entries from the log.
```
Comment (jsha) [inline README.md]:
```suggestion
  * `"getEntries"` for failures to get entries from the log.
```

File: README.md [hunk 4]
```diff
@@ -10,11 +10,368 @@
...
+  * `"checkEntires"` for failures to check unseen certs against the returned new
```
Comment (jsha) [inline README.md]:
```suggestion
  * `"checkEntries"` for failures to check unseen certs against the returned new
```

File: README.md [hunk 5]
```diff
@@ -10,11 +10,368 @@
...
+  certificate to use as the issuer for certificates generated for submitting
+  to logs.
+
+* **inclusionConfig** global configuration related to checking that certificates
+  issued periodically by `ct-woodpecker` were included in the monitored logs.
+
+  * **interval** - a duration string describing the time period between attempts
+    to check unseen certificates for inclusion.
+
+  * **maxGetEntries** - the maximum number of cert entries to ask a log for.
```
Comment (jsha) [inline README.md]: This is not quite right. Instead it should be "The maximum number of log entries to process each interval. `ct-woodpecker` will make a series of `get-entries` calls until it gets this many entries or reaches the tree head, and then process all fetched entries."

File: README.md [hunk 6]
```diff
@@ -10,11 +10,368 @@
...
+  This should be set quite high as `ct-woodpecker` will reduce the number of
+  certs it asks for based on what the log will actually provide.
+
+  * **startIndex** - an optional integer specifying the treesize to start
+  checking for inclusion from. This is useful if you start `ct-woodpecker`
+  monitoring against a log that already has a large tree to let `ct-woodpecker`
```
Comment (jsha) [inline README.md]:
```suggestion
  monitoring against a log that already has a large tree, since it lets `ct-woodpecker`
```

File: README.md [hunk 7]
```diff
@@ -10,11 +10,368 @@
...
+  composed of a config object with the following fields:
+
+  * **uri** - the log's URI.
+
+  * **key** - the log's public key (PEM encoded as a single line without the PEM
+  header/footer).
+
+  * **windowStart** - (optional) for a sharded log the `windowStart` specifies
+  the begin date for the shard's accepted validity window. `ct-woodpecker` will
+  ensure the certificates it generates for this log have a validity period
```
Comment (jsha) [inline README.md]:
```suggestion
  ensure the certificates it generates for this log have a `notAfter`
```

File: README.md [hunk 8]
```diff
@@ -10,11 +10,368 @@
...
+  within the `windowStart` and `windowEnd`
+
+  * **windowEnd** - (optional) for a sharded log the `windowEnd` specifies
+  the end date for the shard's accepted validity window. `ct-woodpecker` will
+  ensure the certificates it generates for this log have a validity period
```
Comment (jsha) [inline README.md]:
```suggestion
  ensure the certificates it generates for this log have a `notAfter`
```

File: README.md [hunk 9]
```diff
@@ -10,11 +10,368 @@
...
+  within the `windowStart` and `windowEnd`
+
+  * **maximum_merge_delay** - the MMD for the log.
+
+  * **submitPreCert** - if true then poisoned precertificates for this log will
```
Comment (jsha) [inline README.md]:
```suggestion
  * **submitPreCert** - if true then precertificates for this log will
```

(`poisoned` is part of the definition of precertificates so we can leave it out)

File: README.md (hunk 4 above)
Comment (cpu) [reply README.md]: How did I make the same typo twice :laughing:

**Comment by cpu** [comment]
@jsha Thanks for the thorough :mag:! As of 56dac13 all of your feedback has been addressed.

**Pull request merged**